connectors/byoc.py

```diff
@@ -297,6 +315,9 @@ def get_filter(self, filter_state="active", domain=DEFAULT_DOMAIN):
             Filter(),
         )
 
+    def to_list(self):
+        return list(self.filtering)
+
```

Comment:
Looks redundant, maybe better to get rid of it?

Reply:
Ah, it's probably better to implement Python's Abstract Base Class that allows certain operations over filtering - maybe just `__iter__` and `__get_item__` and `__len__` will be sufficient - see https://docs.python.org/3/library/collections.abc.html

Reply:
The reason why I added `to_list` to class `Filtering`, is because it was used to update the connector, where a list is required instead of an object:

https://github.com/elastic/connectors-python/blob/59d90d42ba9c01163c0455227d3637ef805d7028/connectors/byoc.py#L118-L134

Similar reason for `to_dict` of class `PipelineSettings`:

https://github.com/elastic/connectors-python/blob/59d90d42ba9c01163c0455227d3637ef805d7028/connectors/byoc.py#L649

I don't intend to make `Filtering` or `PipelineSettings` a collection.